# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1550612003 0
#      Tue Feb 19 21:33:23 2019 +0000
# Node ID a3dfdbb9d8c545ea97de7c46b15d021886f21ee0
# Parent  bbc6b88aacc19c717a2c22500f80704eb5c8a812
# Parent  3d0f687e4a21795df7f9ffc189fa4214d74cd8bc
maint: Merge stable to default.

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -107,20 +107,21 @@ namespace octave
     if (! m_help_engine->setupData())
       {
 #ifdef ENABLE_DOCS
         QMessageBox::warning (this, tr ("Octave Documentation"),
                               tr ("Could not setup the data required for the\n"
                                   "documentation viewer. Only help texts in\n"
                                   "the Command Window will be available."));
 #endif
-        if (m_help_engine)
-          delete m_help_engine;
-        m_help_engine = 0;
-        return;
+
+        disconnect (m_help_engine, 0, 0, 0);
+
+        delete m_help_engine;
+        m_help_engine = nullptr;
       }
 
     // The browser
     QWidget *browser_find = new QWidget (this);
     m_doc_browser = new documentation_browser (m_help_engine, browser_find);
     connect (m_doc_browser, SIGNAL (cursorPositionChanged (void)),
              this, SLOT(handle_cursor_position_change (void)));
 
